### ISSUE TITLE
[osh] echo -E "a\nb"

### ISSUE THREAD

**Issue by bar-g** [oilshell/oil#1843]
Bash also parses `echo -E`.

man bash:  "The -E option disables the  interpretation  of  these escape  characters,  even  on  systems  where they are interpreted by default."
```
$ echo -E "a\nb"
a\nb
$ echo -E -eenEEnneeeE -nnnee 'ALL-PERMUTATION-OPTS-TRIMMED\n'
ALL-PERMUTATION-OPTS-TRIMMED

```

```
osh-0.20.0$ echo -E 'a\nb'
-E a\nb
```

**Comment by bar-g** [comment]
Maybe simplest to just fixing the root of reference to be precise and prevent surprises, i.e. cover all flags by renaming `simple_echo` into more the specific `echo_flags`.

I'll add `-E` to the doc to close this, otherwise I think it's covered by #1772

**Comment by andychu** [comment]
Yeah we should just fix this rather than documenting anything special 

If you are able to get the build process working in https://github.com/oilshell/oil/wiki/Contributing (most people have lately), it should be pretty easy to add a spec test

This is how I find where to put tests

```
$ fgrep 'echo -e' spec/*.test.sh

spec/bugs.test.sh:echo -e 'echo one \0 echo two' > tmp.sh
spec/builtin-io.test.sh:#### echo -e backslashes
spec/builtin-io.test.sh:echo -e \\
spec/builtin-io.test.sh:echo -e '\'
spec/builtin-io.test.sh:echo -e '\\'
spec/builtin-io.test.sh:echo -e "\\"

...
```

So we can test `echo -E` in `spec/builtin-io.test.sh`, which is run by `test/spec.sh builtin-io`

**Comment by bar-g** [comment]
Ok, I already changed the toc-ysh pull requests for echo and write to let their short descriptions focus primarily on what good they do. Instead of what they don't do (-e -n -E dropping pitfall).
